docs/mllib-feature-extraction.md

```diff
@@ -528,6 +528,27 @@ val transformedData2 = parsedData.map(x => transformer.transform(x))
 
 {% endhighlight %}
 </div>
+
+<div data-lang="java">
+{% highlight java %}
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.mllib.feature.ElementwiseProduct;
+import org.apache.spark.mllib.linalg.Vector;
+import org.apache.spark.mllib.linalg.Vectors;
+
+// Create some vector data; also works for sparse vectors
+JavaRDD<Vector> data = sc.parallelize(Arrays.asList(Vectors.dense(1.0, 2.0, 3.0),Vectors.dense(4.0, 5.0, 6.0)));
+
+Vector transformingVector = Vectors.dense(0.0, 1.0, 2.0);
+ElementwiseProduct transformer = new ElementwiseProduct(transformingVector);
+
+// Batch transform and per-row transform give the same results:
+JavaRDD<Vector> transformedData = transformer.transform(data);
+JavaRDD<Vector> transformedData2 = data.map(transformer::transform);
```

Comment:
This is Java 8-specific; I'm not sure other examples "require" java 8 yet? The example isn't quite the same as the Scala one either as it doesn't load the same data file.